fog/view/enclave/impl/src/lib.rs

```diff
@@ -845,3 +885,107 @@ mod get_block_data_tests {
+            let last_known_block_cumulative_txo_count = last_known_block_count * 2;
+            let query_response = create_query_response(
+                last_known_block_count,
+                last_known_block_cumulative_txo_count,
+            );
+            let block_range = BlockRange::new(i as u64, last_known_block_count);
+            let decrypted_query_response = DecryptedMultiViewStoreQueryResponse {
+                query_response,
+                block_range,
+            };
+            decrypted_query_responses.push(decrypted_query_response);
+        }
+
+        let last_response = decrypted_query_responses
+            .last()
+            .expect("Couldn't get last decrypted query response");
+        let expected_last_known_block_count = last_response.query_response.last_known_block_count;
+        let expected_last_known_block_cumulative_txo_count = last_response
+            .query_response
+            .last_known_block_cumulative_txo_count;
+
+        let result = get_last_known_data(&decrypted_query_responses);
+
+        assert_eq!(
+            result.last_known_block_count,
+            expected_last_known_block_count
+        );
+        assert_eq!(
+            result.last_known_block_cumulative_txo_count,
+            expected_last_known_block_cumulative_txo_count
+        );
+    }
+
+    #[test]
+    fn same_last_known_block_counts() {
+        const STORE_COUNT: usize = 4;
+        const LAST_KNOWN_BLOCK_COUNT: u64 = 100;
+        const LAST_KNOWN_BLOCK_CUMULATIVE_TXO_COUNT: u64 = 1000;
+
+        let mut decrypted_query_responses = Vec::with_capacity(STORE_COUNT);
+        for i in 0..STORE_COUNT {
+            let end_block_count = ((i + 1) * 25) as u64;
+            let query_response = create_query_response(
+                LAST_KNOWN_BLOCK_COUNT,
+                LAST_KNOWN_BLOCK_CUMULATIVE_TXO_COUNT,
+            );
+            let block_range = BlockRange::new(i as u64, end_block_count);
+            let decrypted_query_response = DecryptedMultiViewStoreQueryResponse {
+                query_response,
+                block_range,
+            };
+            decrypted_query_responses.push(decrypted_query_response);
+        }
+
+        let last_response = decrypted_query_responses
+            .last()
+            .expect("Couldn't get last decrypted query response");
+        let expected_last_known_block_count = last_response.query_response.last_known_block_count;
+        let expected_last_known_block_cumulative_txo_count = last_response
+            .query_response
+            .last_known_block_cumulative_txo_count;
+
+        let result = get_last_known_data(&decrypted_query_responses);
+
+        assert_eq!(
+            result.last_known_block_count,
+            expected_last_known_block_count
+        );
+        assert_eq!(
+            result.last_known_block_cumulative_txo_count,
+            expected_last_known_block_cumulative_txo_count
+        );
```

Comment:
:wrench: might consider expecting the explicit values instead of programmatically looking up in this case
```suggestion
        assert_eq!(
            result.last_known_block_count,
            LAST_KNOWN_BLOCK_COUNT
        );
        assert_eq!(
            result.last_known_block_cumulative_txo_count,
            LAST_KNOWN_BLOCK_CUMULATIVE_TXO_COUNT
        );
```

Reply:
Done.